zerver/lib/actions.py

```diff
@@ -5344,6 +5344,7 @@ def update_scheduled_email_notifications_time(
     )
 
 
+@transaction.atomic
```

Comment:
Wouldn't it make sense to put this only around this little bit of code?
```
    # TODO: Move these database actions into a transaction.atomic block.
    user_profile.save(update_fields=[setting_name])
    if setting_name in UserProfile.notification_setting_types:
        # Prior to all personal settings being managed by property_types,
        # these were only created for notification settings.
        #
        # TODO: Start creating these for all settings, and do a
        # backfilled=True migration.
        RealmAuditLog.objects.create(
            realm=user_profile.realm,
            event_type=RealmAuditLog.USER_SETTING_CHANGED,
            event_time=event_time,
            acting_user=acting_user,
            modified_user=user_profile,
            extra_data=orjson.dumps(
                {
                    RealmAuditLog.OLD_VALUE: old_value,
                    RealmAuditLog.NEW_VALUE: setting_value,
                    "property": setting_name,
                }
            ).decode(),
        )
```
(whatever we do, we can remove the TODO comment)

if we do, we should probably also just move up the `Draft.objects.filter(user_profile=user_profile).delete()` operation up here to be included in the transaction

Reply:
Do we not want `clear_scheduled_emails` and `update_scheduled_email_notifications_time` in the transaction ?

Reply:
I think we probably do want those in the transaction, yeah, since they are coupled database updates.